### ISSUE TITLE
[SUPPORT]  Questions regarding Cache data clearing using CAD.

### ISSUE THREAD

**Issue by HOLIKADEW** [mrdokenny/Cookie-AutoDelete#1643]
### Acknowledgements

- [x] I acknowledge that I have read the above items

### Please describe what you need support with

Okay, so let me tell you how I want to use the Cookie AutoDelete addon. I basically want each tab to work like a private tab. So, suppose I go to a random website; after I close the tab after 15 seconds, I want CAD to auto-delete everything related to the website, not only limited to just cookies but also local side data and other stuff from my browser, except when I whitelist it by clicking [this](https://imgur.com/a/Qx3BZfl)  

For now, I just downloaded the addon, enabled auto-clearing cookies in the settings, and disabled notifications. I’ve been using it like this for months.

I thought it should delete everything as soon as I leave the tab. Recently, I logged into the TrueCaller website, didn’t make any rules or whitelist it in the addon, closed the tab, and when I came back after 5 minutes, I wasn’t logged out. So, I realized that the addon only deletes cookies, not the site data or other stuff. Also, when I visited my browser's cookies and site data page, I saw many websites still had their local data stored, which I don’t want. [see screenshot](https://imgur.com/a/hb15AXs) also btw how is time.is able to store one cookie into my browser? I don't have any ruleset for the website [see](https://imgur.com/a/WCs0Oy4)  

So, my question is, how can I make the CAD addon delete all cookies along with all the site data when I close the tab, except for the ones I whitelist from [here](https://imgur.com/a/Qx3BZfl)? Don’t you think it should automatically do this? Because when I whitelist a site, it asks me (with a checkbox) if it should keep the cache and other stuff. So, shouldn’t it just delete the data from non-whitelisted sites otherwise, anyway?

In the addon settings, I found a section called "Other Browsing Data Cleanup Options." Under that, it says "Check to delete cache and other stuff." Should check [them](https://imgur.com/a/jHPNQYQ ) all? And if I do, will CAD delete the site data for all sites every time, regardless of the whitelist? And every time I visit a website, will it download all those caches and stuff again? I mean, Is the whitelist only for cookies, or does it clear everything like cache except for cookies? I want the whitelist to keep everything for a site, and for websites that aren’t whitelisted, I want it to clear everything.

It’s also very confusing why we have [that](https://imgur.com/a/jHPNQYQ) option when the whitelist already asks if we want to keep the site data or not. Shouldn’t it just delete the site data for non-whitelisted sites anyway? Why do we have that "Other Browsing Data Cleanup Options" section?

So basically I want the add-on to delete everything single thing, not just the cookies about a website unless I whitelist it myself.

**Comment by kennethtran93** [comment]
> In the addon settings, I found a section called "Other Browsing Data Cleanup Options." Under that, it says "Check to delete cache and other stuff." Should check [them](https://imgur.com/a/jHPNQYQ) all? And if I do, will CAD delete the site data for all sites every time, regardless of the whitelist? And every time I visit a website, will it download all those caches and stuff again? I mean, Is the whitelist only for cookies, or does it clear everything like cache except for cookies? I want the whitelist to keep everything for a site, and for websites that aren’t whitelisted, I want it to clear everything.
> 
> It’s also very confusing why we have [that](https://imgur.com/a/jHPNQYQ) option when the whitelist already asks if we want to keep the site data or not. Shouldn’t it just delete the site data for non-whitelisted sites anyway? Why do we have that "Other Browsing Data Cleanup Options" section?

The 'Other Browsing Data Cleanup Option' pertains to the fact that this web extension's primary function is to delete cookies.

As you noted, the options found in the settings section is a **global** enable/disable setting to determine whether or not to even touch that specific data.  The first setting in that group `Clean Existing Data for Newly Enabled Browsing Data Types` will determine whether or not to clear that data type in its entirety or not once you enable cleanup of those data types using those remaining check boxes within that section.  That being said, once enabled, it will follow your list configuration to remove said data type.  The options within each domain configuration will still show all data types in case you wish to make adjustments even when it is disabled so that it would take effect once the global setting has it enabled.

**Comment by HOLIKADEW** [comment]
I still don’t understand it :( @kennethtran93 
Okay, so all I want to do is clear all data from every website I visit, except for the ones on the whitelist I set from [here](https://imgur.com/a/qHVy8tR). What settings am I supposed to configure?  

By the way, I tried checking everything under [Other Browsing Data Cleanup Options](https://imgur.com/a/7DjsfPK) in the add-on settings, but I still get logged out of some websites like Discord, even though I’ve whitelisted Discord [like this](https://imgur.com/a/8qTOYxT).  

Basically, I want every tab to act like a private tab, deleting all data from the website except for the ones I whitelist. I’d like to whitelist a site by clicking on the add-on in the toolbar and then adding `website.com` and `*.website.com` to the whitelist by clicking on that button that says “whitelist”.

**Comment by Arjun127790** [comment]
Correct me if I'm wrong I think what he means is that enabling those options for cleanup under CAD settings will cleanup the selected data options with the exception of the whitelist. The only question I have for @kennethtran93 is whether enabling one of the options under "Other Browsing Data Cleanup Options" applies only to auto cleanup or manual or both?

**Comment by kennethtran93** [comment]
> Correct me if I'm wrong I think what he means is that enabling those options for cleanup under CAD settings will cleanup the selected data options with the exception of the whitelist. The only question I have for [@kennethtran93](https://github.com/kennethtran93) is whether enabling one of the options under "Other Browsing Data Cleanup Options" applies only to auto cleanup or manual or both?

Depends on your configuration it can apply to both auto and manual cleanup.

At this state, the data _may_ still be kept is because we have yet to fully implement the newer cookie container/management in Firefox.
